### PR TITLE
TOTP Updates

### PR DESCRIPTION
**Pull request by davidlehn**
- Return TOTP params for use when URL or QR Code won't work.
- Use otplib.authenticator to verify.
- Switch from `totpToken` to generic `challenge`.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -400,7 +404,7 @@ api.verify = async ({
   } else if(type === 'totp') {
     // test the totpToken provided to the API against the secret drawn from
     // the accounts database
-    verified = totp.verify({token: totpToken, secret: token.secret});
+    verified = authenticator.verify({token: challenge, secret: token.secret});
```
Comment (mattcollier) [inline lib/index.js]: Why was this changed from `totp` to `authenticator`? It's unclear to me what this implies, but the documentation does say to change it to suite your use case.  https://www.npmjs.com/package/otplib#in-nodejs

Comment (davidlehn) [reply lib/index.js]: It wasn't working with 'totp', but did with 'authenticator'.  Perhaps the defaults are different between the two?

File: lib/index.js [hunk 2]
```diff
@@ -195,14 +195,19 @@ api.set = async ({
...
+    // TOTP params
+    // algorithm, digits, period are defaults for otplib 'authenticator'
     rValue.secret = secret;
+    // FIXME: use readable email and service name
+    rValue.label = email || account;
+    // a service name, `undefined` is OK if `serviceId` was not passed in
+    rValue.service = serviceId;
+    rValue.algorithm = 'SHA1';
+    rValue.digits = 6;
+    rValue.period = 30;
```
Comment (dlongley) [inline lib/index.js]: Add TODO to make algorithm, digits, and period bedrock.config options.
